compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1387,14 +1387,16 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                                 }
                                 None
                             }
+                            // Ignore `use` syntax since that is not valid in objects.
+                            GenericBound::Use(..) => None,
```

Comment:
we could emit the error here instead of in validation.

Edit Hmm I guess that would allow using it in cfged out code tho, nevermind

Reply:
well we don't even parse `dyn use<>` currently. I will delay a bug here.